proposals/0008-c-interface-compiler-library.md

```diff
@@ -996,21 +998,535 @@ if (compileResult->HasOutput(DXC_OUT_OBJECT)) {
 
 ```
 
-## Alternatives considered for supporting legacy toolchains
+## Architecture
+This section describes the surface area calling pattern choices for the
+c-style api.
+
+### Why have a library at all?  Why not just run clang.exe yourself?
+Before diving into the library details we should discuss why we even need a
+library at all.  Why can't clients just execute clang with some options
+and compile their shaders? They can and will. Running clang.exe is a mainline
+supported scenario but we need more to support additional features like
+Include Handlers which are supported by the DXC COM-base DXC library today.
+
+[Include Handlers](#include-handlers) are a way for clients to hook into the
+compilation process and load include dependencies on demand as needed.
+
+### Where should this library be built?
 
-### Alternative 1: Open Source wrapper code for IDxcCompiler interface
+TODO: figure this out
 
-This approach involves creating wrapper code released on
-https://github.com/microsoft that can be included into existing toolchain
-projects. This wrapper implementation will be a drop-in match to all of the
-existing public interfaces allowing easy adoption to using clang.
+### What does the new library design look like?
+The clang compiler based library is c-style only.  It will follow a create/
+destroy pattern for any objects that require lifetime management.
+
+#### Library creation and lifetime
+Instances of the compiler library are created using
+clang_createCompilerLibrary(). Library instances must be destroyed
+using clang_disposeCompilerLibrary().
+
+```c++
+/**
+* An opaque type representing the compiler library.
+*/
+typedef void* CompilerLibraryInstance;
+
+/**
+* Creates an instance of a compiler. Compiler instances must be destroyed by
+* calling clang_disposeCompilerLibrary.
```

Comment:
Yes.  I adopted the create/dispose pattern from that example.  I am open to a different one which could come out as a result of a more public discussion.

Reply:
To be clear, I'm happy with following this convention.  My feedback here is that we should explicitly state which convention we're following so that we can bear this in mind while reviewing it.

Reply:
Sounds good me.  I like it as well and matching an existing pattern is preferred too.